index.js

```diff
@@ -190,7 +180,7 @@ module.exports = Hermes;
 Hermes.prototype.publish = function (queueName, data) {
   /*jshint maxcomplexity:7 */
   debug('hermes publish', queueName, data);
-  if (!~this.queues.indexOf(queueName)) {
+  if (!~this.opts.queues.indexOf(queueName)) {
     throw new Error('attempting to publish to invalid queue: '+queueName);
   }
```

Comment:
this condition seems to be repeated a few times - maybe a helper function would be good? `Hermes.prototype._assertValidQueueName(queueName)` or something?